tachyon/math/base/big_int.h

```diff
@@ -227,6 +294,63 @@ struct ALIGNAS(internal::LimbsAlignment(N)) BigInt {
     return ret;
   }
 
+  // Constructs a BigInt value from a given array of bytes interpreted in
+  // big-endian order. The method processes each byte of the input, packs them
+  // into 64-bit limbs, and then sets these limbs in the resulting BigInt. If
+  // the system is big-endian, adjustments are made to ensure correct byte
+  // ordering.
+  template <size_t Size>
+  constexpr static BigInt FromBytesBE(const std::array<uint8_t, Size>& bytes) {
+    BigInt ret;
+    size_t byte_idx = 0;
+    size_t limb_idx = 0;
+    uint64_t limb = 0;
+    FOR_FROM_BIGGEST(i, 0, bytes.size()) {
+      reinterpret_cast<uint8_t*>(&limb)[byte_idx++] = bytes[i];
+      bool set = byte_idx == kLimbByteNums;
+#if ARCH_CPU_BIG_ENDIAN
+      set |= (i == bytes.size() - 1);
+#else
+      set |= (i == 0);
+#endif
+      if (set) {
+        ret.limbs[limb_idx++] = limb;
+        limb = 0;
+        byte_idx = 0;
+      }
+    }
+    return ret;
+  }
+
+  // Constructs a BigInt value from a given InlinedVector of bytes interpreted
+  // in big-endian order. The method processes each byte of the input, packs
+  // them into 64-bit limbs, and then sets these limbs in the resulting BigInt.
+  // If the system is big-endian, adjustments are made to ensure correct byte
```

Comment:
Do you assume the platform is little-endian or means the opposite case of function?

```suggestion
  // the system is little-endian, adjustments are made to ensure correct byte
```